base/shell/explorer/traydeskbtn.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Show Desktop tray button implementation
+ * COPYRIGHT:   Copyright 2006-2007 Thomas Weidenmueller <w3seek@reactos.org>
+ *              Copyright 2018-2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ *              Copyright 2023 Ethan Rodensky <splitwirez@gmail.com>
+ */
+
+#include "traydeskbtn.h"
+#include <limits.h>
+
+#define IDI_SHELL32_DESKTOP 35
+#define IDI_IMAGERES_DESKTOP 110
+
+#define SHOW_DESKTOP_TIMER_ID 999
+#define SHOW_DESKTOP_TIMER_INTERVAL 200
+
+CTrayShowDesktopButton::CTrayShowDesktopButton() :
+    m_nClickedTime(0),
+    m_inset({2, 2}),
+    m_icon(NULL),
+    m_highContrastMode(FALSE),
+    m_drawWithDedicatedBackground(FALSE),
+    m_bHovering(FALSE),
+    m_bPressed(FALSE),
+    IsHorizontal(FALSE)
+{
+}
+
+BOOL
+CTrayShowDesktopButton::GetTaskbar(OUT HWND* taskbarWnd)
+{
+    *taskbarWnd = NULL;
+    auto parent = GetParent();
+    if (!parent)
+        return FALSE;
+
+    parent = parent.GetParent();
+    if (!parent)
+        return FALSE;
+
+    HWND wnd = parent.m_hWnd;
+    *taskbarWnd = wnd;
+    return wnd != NULL;
+}
+
+INT CTrayShowDesktopButton::WidthOrHeight() const
+{
+    if (IsThemeActive() && !m_highContrastMode)
+    {
+        if (m_drawWithDedicatedBackground)
+        {
+            if (GetSystemMetrics(SM_TABLETPC))
+            {
+                //TODO: DPI scaling - return logical-to-physical conversion of 24, not fixed value
```

Comment:
If you didn't want to hardcode a fixed value here, what would have to be done?

Reply:
I'm not sure of the specifics - I was just going off of what you told me. (see DMs on Discord)